src/server/scripts/Outland/GruulsLair/boss_gruul.cpp

```diff
@@ -122,14 +126,19 @@ class boss_gruul : public CreatureScript
                     events.DelayEvents(8001);
                     events.ScheduleEvent(EVENT_GROUND_SLAM, 60000);
                     events.ScheduleEvent(EVENT_SHATTER, 8000);
+                    me->SetControlled(true, UNIT_STATE_ROOT);       // UNIT_STATE_STUNNED would prevent finishing the cast of Ground Slam
                     break;
                 case EVENT_SHATTER:
                     Talk(SAY_SHATTER);
+                    me->SetControlled(false, UNIT_STATE_ROOT);
                     me->CastSpell(me, SPELL_SHATTER, false);
                     break;
             }
 
-            DoMeleeAttackIfReady();
+            if (!me->HasUnitState(UNIT_STATE_ROOT))             // Since we can not just use the UNIT_STATE_STUNNED flag on Gruul,
+            {
+                DoMeleeAttackIfReady();                         // we need to prevent his autohits like this while he is rooted
+            }
```

Comment:
```suggestion
            if (!me->HasUnitState(UNIT_STATE_ROOT))
            {
                DoMeleeAttackIfReady();
            }
```
we deduce how a boss works by looking at the script, no need for these comments unless totally unexpected functionality or complex function.